src/handlers/telegramManager/TelegramManager.ts

```diff
@@ -0,0 +1,147 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Neil Enns. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { promises as fsPromise } from 'fs';
+import * as JSONC from 'jsonc-parser';
+import TelegramBot from 'node-telegram-bot-api';
+
+import * as log from '../../Log';
+import telegramManagerConfigurationSchema from '../../schemas/telegramManagerConfiguration.schema.json';
+import validateJsonAgainstSchema from '../../schemaValidator';
+import Trigger from '../../Trigger';
+import IDeepStackPrediction from '../../types/IDeepStackPrediction';
+import ITelegramManagerConfigJson from './ITelegramManagerConfigJson';
+
+let isEnabled = false;
+let telegramBot: TelegramBot;
+
+/**
+ * Takes a path to a configuration file and loads all of the triggers from it.
+ * @param configFilePath The path to the configuration file
+ */
+export async function loadConfiguration(configFilePath: string): Promise<void> {
+  const rawConfig = await readRawConfigFile(configFilePath);
+
+  if (!rawConfig) {
+    return;
+  }
+
+  const telegramConfigJson = parseConfigFile(rawConfig);
+
+  if (!(await validateJsonAgainstSchema(telegramManagerConfigurationSchema, telegramConfigJson))) {
+    // This throws an error instead of allowing startup to proceed since the assumption is
+    // if the user specified a configuration file they actually do want MQTT enabled
+    // and running. It would be bad if this continued to run with MQTT disabled
+    // and the user thought MQTT events were getting sent when they weren't.
+    throw new Error("[Telegram Manager] Invalid configuration file.");
+  }
+
+  log.info("Telegram manager", `Loaded configuration from ${configFilePath}`);
+
+  telegramBot = new TelegramBot(telegramConfigJson.botToken, {
+    filepath: false,
+  });
+
+  isEnabled = true;
+}
+
+export async function processTrigger(
+  fileName: string,
+  trigger: Trigger,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  predictions: IDeepStackPrediction[],
+): Promise<TelegramBot.Message[]> {
+  if (!isEnabled) {
+    return [];
+  }
+
+  // It's possible to not set up an mqtt handler on a trigger or to disable it, so don't
+  // process if that's the case.
+  if (!trigger?.telegramConfig?.enabled) {
+    return [];
+  }
+
+  log.info("Telegram Manager", `${fileName}: Publishing event to ${trigger.mqttConfig.topic}`);
```

Comment:
Wrong log message. Should be the chatId, not an MQTT topic name. Again bad copy paste.

Reply:
Honestly this shouldn't be here at all since `sendTelegramMessage()` does its own logging.

Reply:
Fixed